File: app/repositories/NoteRepository.java
package repositories;

import java.util.List;

import models.Note;

import org.springframework.data.repository.PagingAndSortingRepository;

public interface NoteRepository extends PagingAndSortingRepository<Note, String> {

	public List<Note> findByNotebookId(String notebookId);
	
}

File: server/note/src/main/kotlin/fr/acle/notello/note/repository/NoteRepository.kt
package fr.acle.notello.note.repository

import fr.acle.notello.note.model.Note
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository


interface NoteRepository : PagingAndSortingRepository<Note, String> {
    fun findByNotebookId(notebookId: String): List<Note>
    fun findByNotebookId(notebookId: String, pageable: Pageable): Page<Note>
}
